### ISSUE TITLE
Property name starting from "data" is recognized as data-* attribute

### ISSUE THREAD

**Issue by web-padawan** [Polymer/polymer-linter#68]
I get this warning in several cases:
```
[set-unknown-attribute] - data-* attributes must be accessed as attributes. i.e. you must write:  data-for-export$="{{dataForExport}}"
```

As of now, Polymer 1.x accepts such properties, though (and those don't get reflected into attributes, neither are they being put into `dataset`).

Would it be disallowed by 2.0 release? Should we avoid such names? 
If so, having this documented would be nice.
If not, I'd expect not to see those warnings at all.

**Comment by rictic** [comment]
So, this is in that category of legal but suspicious code, as the user may be trying to interact with `dataset`, and would be surprised when that doesn't happen.

Most lint warnings do have some false negatives, as there's a lot of wrong code that won't throw.

OTOH, it may be more common to have dataFoo properties than to intentionally use the `dataset` field. I'm not sure that it actually sees much real world use?

Hm, oh! If there's a known property like `dataForExport` that you're setting, then we should not warn. Only if it's an unknown property and it starts with `data-` should we give the special warning about `data-*` attributes.

Ok, cool, yeah that should work.

Thanks for filing these bugs btw!
